src/random.cpp

```diff
@@ -191,6 +191,7 @@ void GetDevURandom(unsigned char *ent32)
     do {
         ssize_t n = read(f, ent32 + have, NUM_OS_RANDOM_BYTES - have);
         if (n <= 0 || n + have > NUM_OS_RANDOM_BYTES) {
+            close(f);
```

Comment:
Yeah, I can see that. I didn't want to add more code than necessary here, but personally I would have prefered a proper resource handler, and a better exit from GetDevURandom.
As it stands now, GetDevURandom still depend on RandFailure to actually exit the program completely.